### PR TITLE
 Fix Punycode problem when register

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.4.x
| Description?  | Since domain are allowed in UTF-8, some don't respect the RFC 2822, so to prevent error with this version of SwiftMailer (could not update due to PHP requirements), we must convert domain name in Punycode. Thanks Wikipédia: https://en.wikipedia.org/wiki/Punycode.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-4398
| How to test?  | Create an account on firefox with `test@voilà.fr` or `test@домен.com`.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9080)

### CODE REVIEW

File: classes/Mail.php [hunk 1]
```diff
@@ -49,22 +49,53 @@ class MailCore extends ObjectModel
...
+            ],
+        ],
+    ];
 
     const TYPE_HTML = 1;
     const TYPE_TEXT = 2;
     const TYPE_BOTH = 3;
 
+    const METHOD_SMTP = 2;
+    const METHOD_DISABLE = 3;
```
Comment (tomlev) [inline classes/Mail.php]: please explain this const, I was required to read code to understand it

**Review by tomlev — CHANGES_REQUESTED** [review]

File: classes/Mail.php (hunk 1 above)
Comment (PierreRambaud) [reply classes/Mail.php]: Done!

**Review by tomlev — DISMISSED** [review]

File: classes/Mail.php [hunk 2]
```diff
@@ -249,40 +304,48 @@ public static function send(
...
+
                     return false;
                 }
-                $message->addBcc($addr);
+
+                $message->addBcc(self::toPunycode($addr));
             }
         } elseif (isset($bcc)) {
-            $message->addBcc($bcc);
+            $message->addBcc(self::toPunycode($bbc));
```
Comment (eternoendless) [inline classes/Mail.php]: Watch out, I think it's `$bcc` instead of `$bbc` (Thank you Codacy)

**Review by eternoendless — CHANGES_REQUESTED** [review]

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
Ok to merge once the tests are green

**Comment by eternoendless** [comment]
Thank you @PierreRambaud

**Comment by ilyasovivan** [comment]
THIS IS NOT FIXED. SEE THIS FOR MORE INFO: https://github.com/PrestaShop/PrestaShop/issues/11588

**Pull request merged**